platform/platform-impl/src/com/intellij/openapi/vfs/newvfs/RefreshSessionImpl.java

```diff
@@ -122,10 +127,15 @@ void scan() {
         ((LocalFileSystemImpl)fs).markSuspiciousFilesDirty(workQueue);
       }
 
-      long t = 0;
+      long t = System.currentTimeMillis();
+      String refreshSessionInfo = String.format("work queue size: %s , file types: %s", workQueue.size(),
```

Comment:
we cannot do this conditional, unfortunately, since we don't know how long session might take and queue won't be there by the time we done. We can accumulate processed entries but it might be worse than compute this unconditionally

Reply:
oh, my bad, it is indeed immutable. Moving under condition then

Reply:
well... it is not exactly immutable and more items can be added since we started session in theory (depending on in which order launch() and scan() called). But since there is no way to delete it is ok to use queue afterwards.

Reply:
It is effectively immutable. Once `scan()` is called, the set of roots this session will process is fixed.; any new roots are simply ignored. So yes, it is perfectly okay to use the `workQueue` variable anywhere in the method.